[rizkidelta] IP

rizkidelta opened the pull request:
**DukeV2**

> Simple scheduler for simple people

**Duke is a scheduler able to keep track of your**
- Deadlines
- Todos
- Events

**Benefits of using me:**
1. Nice GUI
2. Able to use on both cmd and GUI
3. Save tasks across sessions

**Commands**
```
list
mark <task num>
unmark <task num>
delete <task num>
find <word>
bye
todo <desc>
deadline <desc> /by <dd/MM/yyyy> <HHmm>
event <desc> /from <dd/MM/yyyy> <HHmm> /to <dd/MM/yyyy> <HHmm>
```
**Goals:**
- [ ] Do weekly task on time :upside_down_face:

**Download**
Download [here](https://github.com/rizkidelta/ip/releases/tag/A-Jar)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

rizkidelta committed:
Level 0. Rename, Greet, Exit

rizkidelta committed:
Level 1. Echo

rizkidelta committed:
Level 2. Add, List

rizkidelta committed:
Level 3. Mark as Done

rizkidelta committed:
Level 4. ToDos, Events, Deadlines

rizkidelta committed:
Automated Text UI Testing

rizkidelta committed:
Level 5. Handle Errors

rizkidelta committed:
Level 6. Delete

rizkidelta committed:
Added Level 7. Save

Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up. You may hard-code the file name and relative path from the project root

rizkidelta committed:
Merge pull request #1 from rizkidelta/branch-Level-7

Added Level 7. Save

rizkidelta committed:
Added Level 8. Dates and Times

Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

rizkidelta committed:
Merge pull request #2 from rizkidelta/branch-Level-8

Added Level 8. Dates and Times

rizkidelta committed:
Updated Data.txt Location

rizkidelta committed:
A-MoreOOP
Refactor the code to extract out closely related code as classes.

rizkidelta committed:
Merge pull request #3 from rizkidelta/add-gradle-support

Add gradle support

rizkidelta committed:
Updated Data.txt

Data.txt blank as default

rizkidelta committed:
A-JUnit

rizkidelta committed:
A-JavaDoc

Add JavaDoc comments to the code.

rizkidelta committed:
Level 9. Find

Give users a way to find a task by searching for a keyword.

Cedricaca commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,109 @@
+                    case DELETE:
+                        index = Integer.parseInt(inputs[1]) - 1;
+                        Task toDelete = tasks.get(index);
+                        tasks.remove(toDelete);
```
perhaps have tasks.remove() take in the index input and return the deleted task

Cedricaca commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
```
feel like this could be better named. not sure what parts[2], parts[3] supposed to represent

Cedricaca commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+        } else {
+            return type + " | " + status + " | " + description;
+        }
+    }
```
this could possibly be a function of the task object rather than storage

Cedricaca reviewed (commented):
Well written and easy to read code. I dont have major issues with code quality guidelines. Good job!

GERARDJM018 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,109 @@
+
+        //Execute Inputs
+        boolean isExit = false;
+        while (!isExit){
```
```suggestion
        while (!isExit) {
```
Perhaps use whitespace before the open curly bracket (to make the code cleaner)?
I noticed the same issue in several other places too

GERARDJM018 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,109 @@
+                        break;
+                    default:
+                        ui.printError("Command not found!");
+                        break;
```
Shouldn't there be no indentation for `case` clauses when using switch statement (According to the given [Java Coding Standar](https://se-education.org/guides/conventions/java/intermediate.html))?

GERARDJM018 commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+ * The file format used is text-based with specific delimiters for task attributes.
+ *
+ * @author Muhammad Rizki Bayuaji
+ */
```
I like how you give Java Doc Comment to explain the class 👍

GERARDJM018 commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,89 @@
+        return this.description;
+    }
+
+    /**
+     * Marks the task as done.
+     *
+     * @author Muhammad Rizki Bayuaji
+     */
+    public void mark() {
+        isDone = true;
+    }
+
+    /**
+     * Unmarks the task, setting its status to not done.
+     *
+     * @author Muhammad Rizki Bayuaji
+     */
+    public void unMark() {
+        isDone = false;
+    }
```
```suggestion
    /**
     * Marks the task as done.
     *
     * @author Muhammad Rizki Bayuaji
     */
    public void setMark() {
        isDone = true;
    }

    /**
     * Unmarks the task, setting its status to not done.
     *
     * @author Muhammad Rizki Bayuaji
     */
    public void SetUnMark() {
        isDone = false;
    }
```
Perhaps change the name method using the format for "setter methods for boolean variables"?

GERARDJM018 commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline task in a task management application.
+ * A deadline task includes a description and a deadline date and time.
+ * It extends the Task class by adding a deadline-specific time attribute.
+ * The deadline time is managed using the LocalDateTime class.
+ * The input time format is d/M/yyyy HHmm (e.g., 2/12/2019 1800).
+ * If an invalid date format is provided, an error message is displayed.
+ *
+ * @author Muhammad Rizki Bayuaji
+ */
```
Perhaps you can omit some of the details here since you already explained it in more detail in the related method?

GERARDJM018 reviewed (commented):
Looks good. 💯 
There is some feedback/suggestions for certain parts of the code.

rizkidelta committed:
A-CheckStyle

Added checkstyle files to run checkstyle

rizkidelta committed:
Level 10. GUI

rizkidelta committed:
The GUI lacks necessary files for proper functionality.

Without these files, the GUI won't display correctly or provide the intended user experience.

Add the required image file and FTML to enable the GUI.

Adding these files is imperative for the GUI's proper rendering and functionality.

Once these files are added, the GUI will be fully operational, enhancing the usability of the application.

rizkidelta committed:
Merge pull request #5 from rizkidelta/branch-Level-10

The GUI lacks necessary files for proper functionality.

rizkidelta committed:
The system lacks handling for missing files and input length validation, potentially causing errors.

This omission may disrupt the program flow and hinder user experience, necessitating correction.

Employing Java assertions to handle missing files and validate input length ensures robustness and correctness.

Java assertions facilitate proactive error management, enhancing reliability and maintainability.

Integrating Java assertions improves system resilience and user experience.

rizkidelta committed:
Merge pull request #6 from rizkidelta/branch-A-Assertions

The system lacks handling for missing files and input length validation, potentially causing errors.

rizkidelta committed:
Code examination revealed issues with variable and function naming conventions, incorrect indentations, and package names not adhering to lowercase conventions.

These issues impact code readability, maintainability, and adherence to coding standards, necessitating improvement.

Refactoring the code to adhere to proper naming conventions for variables and functions, correcting indentations, and ensuring package names are lowercase.

Following consistent naming conventions and proper indentation improves code clarity and maintainability, while adhering to lowercase package names aligns with standard coding practices.

The refactoring process enhances code quality and readability, making it easier to understand and maintain in the long run.

rizkidelta committed:
Merge pull request #7 from rizkidelta/branch-A-CodeQuality

Code examination revealed issues with variable and function naming co…

rizkidelta committed:
The system now has the ability to recognize and handle duplicate items, such as the same task being added multiple times.

Without addressing this issue, duplicate items could lead to inconsistencies and incorrect data representation within the system.

Implemented a mechanism to check for duplicate tasks before adding them to the list, ensuring data integrity and preventing redundancy.

By checking for duplicates before adding tasks, the system maintains consistency and efficiency, enhancing overall performance and user experience.

With the addition of duplicate item recognition and handling, the system becomes more robust and reliable, providing users with a smoother interaction.

rizkidelta committed:
Merge pull request #8 from rizkidelta/branch-BCD-Extension

The system now has the ability to recognize and handle duplicate item…

rizkidelta committed:
Changed name of bot to Duc

rizkidelta committed:
Add files via upload

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md

In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Update README.md


In the latest update of the README document, we've added concise instructions for inserting images in Markdown and a streamlined 'Getting Started' guide. These enhancements aim to simplify user onboarding and enrich the documentation's clarity and visual appeal.

rizkidelta committed:
Rename ui.PNG to Ui.PNG

rizkidelta committed:
Update README.md

rizkidelta committed:
Rename Ui.PNG to Ui.png

rizkidelta committed:
Update README.md

rizkidelta committed:
Update README.md

rizkidelta committed:
Update README.md

rizkidelta committed:
Update README.md
